src/OpenFeature.Contrib.Providers.AwsAppConfig/AppConfigProvider.cs

```diff
@@ -0,0 +1,212 @@
+        /// <param name="cancellationToken">Cancellation token for async operations</param>
+        /// <returns>Resolution details containing the structured flag value</returns>
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValueAsync(string flagKey, Value defaultValue, EvaluationContext context = null, CancellationToken cancellationToken = default)
+        {
+            var flagValue = await ResolveFeatureFlagValue(flagKey, defaultValue);
+            return new ResolutionDetails<Value>(flagKey, new Value(flagValue));
+        }
+
+        /// <summary>
+        /// Resolves a feature flag value from AWS AppConfig, optionally extracting a specific attribute.
+        /// </summary>
+        /// <param name="flagKey">The feature flag key, which can include an attribute specification in the format "flagKey:attributeKey"</param>
+        /// <param name="defaultValue">The default value to return if the flag or attribute cannot be resolved</param>
+        /// <returns>
+        /// A Value object containing the resolved feature flag value. If the key includes an attribute specification,
+        /// returns the value of that attribute. Otherwise, returns the entire flag value.
+        /// </returns>
+        /// <remarks>
+        /// This method handles two types of feature flag resolution:
+        /// 1. Simple flag resolution: When flagKey is a simple key (e.g., "myFlag")
+        /// 2. Attribute resolution: When flagKey includes an attribute specification (e.g., "myFlag:someAttribute")
+        /// 
+        /// The method first retrieves the complete feature flag configuration and then:
+        /// - For simple flags: Returns the entire flag value
+        /// - For attribute-based flags: Returns the specific attribute value if it exists, otherwise returns the default value
+        /// </remarks>
+        /// <example>
+        /// Simple flag usage:
+        /// <code>
+        /// var value = await ResolveFeatureFlagValue("myFlag", new Value(defaultValue));
+        /// </code>
+        /// 
+        /// Attribute-based usage:
+        /// <code>
+        /// var value = await ResolveFeatureFlagValue("myFlag:color", new Value("blue"));
+        /// </code>
+        /// </example>
+        private async Task<Value> ResolveFeatureFlagValue(string flagKey, Value defaultValue)
+        {
+            var appConfigKey = new AppConfigKey(flagKey);
+
+            var responseString = await GetFeatureFlagsResponseJson(appConfigKey.ConfigurationProfileId);
+
+            var flagValues = FeatureFlagParser.ParseFeatureFlag(appConfigKey.FlagKey, defaultValue, responseString);
+
+            if (!appConfigKey.HasAttribute) return flagValues;
+
+            var structuredValues = flagValues.AsStructure;
+
+            if(structuredValues == null) return defaultValue;
+
+            return structuredValues.TryGetValue(appConfigKey.AttributeKey, out var returnValue) ? returnValue : defaultValue;

```

Comment:
We should return extra information (Flag not Found) in case we are resolving using the default flag.